hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java

```diff
@@ -1466,5 +1476,27 @@ public void setStoragePolicy(Path path, String policyName)
         throws IOException {
       throw readOnlyMountTable("setStoragePolicy", path);
     }
+
+    private FsPermission getMountLinkDefaultPermissions() {
+      return PERMISSION_555;
+    }
+
+    private String getMountLinkUserName() {
+      if (mountLinkUserName != null) {
+        return mountLinkUserName;
+      }
+      String username = ugi.getShortUserName();
+      mountLinkUserName = username;
+      return username;
+    }
+
+    private String getMountLinkGroupName() throws IOException {
+      if (mountLinkGroupName != null) {
+        return mountLinkGroupName;
+      }
+      String groupname = ugi.getPrimaryGroupName();
+      mountLinkGroupName = groupname;
```

Comment:
Instead of setting this, a cleaner way is to declare mountLinkGroupName as a final variable and return ugi.getPrimaryGroupName() here. Same goes for the implementation of getMountLinkUserName()

Reply:
alternatively, just set mountLinkGroupName to ugi.getPrimaryGroupName() in the constructor if the config is not set?

Reply:
Go with the second option, updated.

Reply:
Tip for future - leave the comment unresolved so that the person who asked for the change can resolve it. It makes it easier for the reviewer to see what changed.

Reply:
Actually, for the second option we changed the behavior by setting `mountLinkGroupName` to `ugi.getPrimaryGroupName()` in the constructor, since before we only set it on demand when we call `listStatus()`, `getFileStatus()`, etc... Not sure this behavior change will cause other problems but this failed some unit tests like `testListStatusWithNoGroups` with error "There is no primary group for UGI" when we initialize `InternalDirOfViewFs`. Because in the test we don't expect the error during filesystem initialization.

Reply:
For your first suggested option:
I think the idea of setting `mountLinkGroupName` in this function is to not call `ugi.getPrimaryGroupName()` every time we call this get method. We set `mountLinkGroupName` once if it's null then there's no need to call `ugi.getPrimaryGroupName()` again.